resources/fetching/fetchers/azure/assets_enricher_storage_account.go

```diff
@@ -59,13 +59,35 @@ func (e storageAccountEnricher) Enrich(ctx context.Context, cycleMetadata cycle.
 		errs = append(errs, fmt.Errorf("storageAccountEnricher: error while getting services diagnostic settings: %w", err))
 	}
 
+	storageAccountsSubscriptionsIds := lo.Uniq(lo.Map(storageAccounts, func(item inventory.AzureAsset, i int) string {
+		return item.SubscriptionId
+	}))
```

Comment:
[Optional small improvement]
we could put something like this 
```golang
subIDs := lo.Keys(subscriptions)
```
at, e.g., line 45 and use the `subIDs` slice both in line 46 and in `e.addStorageAccounts`

Reply:
i don't think i can use this as `lo.Keys(subscriptions)` is not the same as the set of storage account subscription ids, which is smaller, as it only consists of the storage account type, not all subscriptions. 

for example, on our dev account the first has a length of 139 and the 2nd has 13